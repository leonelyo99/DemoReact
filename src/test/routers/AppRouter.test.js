import React from 'react';
import { mount, shallow } from 'enzyme';
import { PrivateRoute } from '../../routers/PrivateRoute';
import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from '../../auth/AuthContext';


describe('Pruebas en <Private/>', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }

    test('debe de mostrar el login si no esta autenticado', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar el componente marvel si esta autenticado', () => {
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Leonel'
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        );

        expect(wrapper.find('.navbar').exists()).toBe(true); //si esta el navbar mostro el componente ya que esta logueado
    })
    
})