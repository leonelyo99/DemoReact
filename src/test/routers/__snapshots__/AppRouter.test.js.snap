// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Private/> debe de mostrar el login si no esta autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <form
      onSubmit={[Function]}
    >
      <input
        autoComplete="off"
        className="form-control"
        name="name"
        onChange={[Function]}
        placeholder="Write your name"
        type="text"
        value=""
      />
      <button
        className="btn btn-primary mt-4"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
